### PR TITLE
Support for next 13.5

### LINKED ISSUE #104: Next 13.4.20-canary.20 issue
Any idea how to make the next-router-module work with the next@13.4.20-canary.20 or later?

I'm getting this error:

```
Cannot find module 'next/dist/shared/lib/router-context' from 'node_modules/next-router-mock/dist/MemoryRouterProvider/next-11.js'

    Require stack:
      node_modules/next-router-mock/dist/MemoryRouterProvider/next-11.js
      node_modules/next-router-mock/dist/MemoryRouterProvider/next-13.js
      node_modules/next-router-mock/dist/MemoryRouterProvider/index.js
      src/lib/testHelpers.tsx
```

### PR DESCRIPTION
**Pull request by jantimon**
Next 13.5 renamed a file.

This pr adds:

 - support for 13.5
 - tests for 13.5

FIxes #104

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,8 +18,8 @@
     "test": "jest",
     "test:watch": "jest --watch",
     "test:integration:ci": "npm run test:integration:install && npm run test:integration",
-    "test:integration:install": "(cd test/example-app && npm i); (cd test/next-10 && npm i); (cd test/next-11 && npm i); (cd test/next-12 && npm i); (cd test/next-12.2+ && npm i); (cd test/next-12.latest && npm i); (cd test/next-13 && npm i)",
-    "test:integration": "npm run build && npm run test:exports && npm run test:example-app && npm run test:next-10 && npm run test:next-11 && npm run test:next-12 && npm run test:next-12.2 && npm run test:next-12.latest && npm run test:next-13",
+    "test:integration:install": "(cd test/example-app && npm i); (cd test/next-10 && npm i); (cd test/next-11 && npm i); (cd test/next-12 && npm i); (cd test/next-12.2+ && npm i); (cd test/next-12.latest && npm i); (cd test/next-13 && npm i); (cd test/next-13.5 && npm i)",
+    "test:integration": "npm run build && npm run test:exports && npm run test:example-app && npm run test:next-10 && npm run test:next-11 && npm run test:next-12 && npm run test:next-12.2 && npm run test:next-12.latest && npm run test:next-13 && npm run test:next-13",
```
Comment (scottrippey) [inline package.json:22]:
```suggestion
    "test:integration": "npm run build && npm run test:exports && npm run test:example-app && npm run test:next-10 && npm run test:next-11 && npm run test:next-12 && npm run test:next-12.2 && npm run test:next-12.latest && npm run test:next-13 && npm run test:next-13.5",
```

File: src/MemoryRouterProvider/next-11.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // NOTE: this path works with Next v11.1.0+
```
Comment (scottrippey) [inline src/MemoryRouterProvider/next-11.tsx:1]:
```suggestion
// NOTE: this path works with Next v11.1.0 through v13.4.0+
```

File: test/next-13.5/next-13.5.test.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+// TODO: Officially support Next 13
+// Validate our types are exported correctly:
+import type { createDynamicRouteParser } from "../../dynamic-routes";
+import type { createDynamicRouteParser as _ } from "../../dynamic-routes/next-13";
+import type { MemoryRouterProvider } from "../../MemoryRouterProvider";
+import type { MemoryRouterProvider as __ } from "../../MemoryRouterProvider/next-13";
```
Comment (scottrippey) [inline test/next-13.5/next-13.5.test.ts:6]: We'll need to create another entry-point here too:

```suggestion
import type { MemoryRouterProvider as __ } from "../../MemoryRouterProvider/next-13.5";
```

File: test/next-13.5/next-13.5.test.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      require("../../dynamic-routes");
+    });
+    it("next-router-mock/dynamic-routes/next-13", () => {
+      require("../../dynamic-routes/next-13");
+    });
+    it("next-router-mock/MemoryRouterProvider", () => {
+      require("../../MemoryRouterProvider");
+    });
+    it("next-router-mock/MemoryRouterProvider/next-13", () => {
+      require("../../MemoryRouterProvider/next-13");
```
Comment (scottrippey) [inline test/next-13.5/next-13.5.test.ts:20]:
```suggestion
      require("../../MemoryRouterProvider/next-13.5");
```

**Review by scottrippey — CHANGES_REQUESTED** [review]
Fantastic work, thank you for copying the existing patterns so well!

ToDo:
- [ ] Add `/MemoryRouterProvider/next-13.5/package.json` entrypoint
- [ ] Make updates listed below
- [ ] Update `README.md` with instructions
- [x] Complain to Next.js that they keep making it hard to test `next/router` and `next/link` because they keep moving these files around (done: https://github.com/vercel/next.js/discussions/38248)

File: .changeset/hot-bears-bake.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"next-router-mock": minor
+---
+
+Add support for Next 13.5
```
Comment (scottrippey) [inline .changeset/hot-bears-bake.md:5]: Updated `MemoryRouterProvider` to support Next 13.5

**Review by scottrippey — APPROVED** [review]

**Comment by jantimon** [comment]
nice work! :)

**Review by PizzaPete — APPROVED** [review]

**Review by gmerwan — APPROVED** [review]
Great ✅ .. looking forward to the update 🙏🏻

**Pull request merged**